src/main/java/hudson/plugins/ec2/util/EC2SlaveFactory.java

```diff
@@ -0,0 +1,32 @@
+package hudson.plugins.ec2.util;
+
+import java.io.IOException;
+import java.util.List;
+
+import hudson.model.Descriptor;
+import hudson.model.Node;
+import hudson.plugins.ec2.*;
+import hudson.slaves.NodeProperty;
+import jenkins.model.Jenkins;
+
+public interface EC2SlaveFactory {
+
+    static EC2SlaveFactory getInstance() {
+        EC2SlaveFactory instance = null;
+        for (EC2SlaveFactory implementation : Jenkins.get().getExtensionList(EC2SlaveFactory.class)) {
+            if (instance != null) {
+                throw new IllegalStateException("Multiple implementations of " + EC2SlaveFactory.class.getName()
+                        + " found. If overriding, please consider using ExtensionFilter");
+            }
+            instance = implementation;
+        }
+        return instance;
+    }
+
+    EC2OndemandSlave createOnDemandSlave(String name, String instanceId, String description, String remoteFS, int numExecutors, String labelString, Node.Mode mode, String initScript, String tmpDir, List<? extends NodeProperty<?>> nodeProperties, String remoteAdmin, String jvmopts, boolean stopOnTerminate, String idleTerminationMinutes, String publicDNS, String privateDNS, List<EC2Tag> tags, String cloudName, boolean useDedicatedTenancy, int launchTimeout, AMITypeData amiType, ConnectionStrategy connectionStrategy, int maxTotalUses)
```

Comment:
Would it make more sense to use some sort of config class object here? Having more than a couple method parameters in Java gets really hard to deal with.

Reply:
Yeah I suppose. I just went with the lowest effort change to just retain the current parameters.

But how about a config object with a builder pattern that throws an exception for missing parameters?

Reply:
That sounds like a great way to approach it!

Reply:
I pushed a change here.

I decided against throwing Exceptions since I don't really know if some of those parameters should in fact be nullable.